polaris-patterns/src/components/DatePicker/DatePicker.tsx

```diff
@@ -1,9 +1,21 @@
 import React from 'react';
-import {Weekday, useI18n} from '@shopify/react-i18n';
+import {Weekday, useI18n, I18nContext} from '@shopify/react-i18n';
 import type {DatePickerProps} from '@shopify/polaris';
 import {DatePicker as PolarisDatePicker} from '@shopify/polaris';
 
+import {useShopifyI18nManager} from '../PolarisPatternsProvider';
+
 export function DatePicker(props: DatePickerProps) {
+  const manager = useShopifyI18nManager();
+
+  return (
+    <I18nContext.Provider value={manager}>
+      <DatePickerComponent {...props} />
+    </I18nContext.Provider>
+  );
+}
+
+export function DatePickerComponent(props: DatePickerProps) {
```

Comment:
I'm using a HOC to wrap the individual component that needs a the `@shopify/react-i18n` library since asking consuming apps to wrap their whole application would cause a translation conflict.

This might be a good pattern to establish in this package? if the exported component needs a specific library that wouldn't need an app wide context, we can wrap the library provider around that specific component that needs it